cypress/utils/deployLSPContract.js

```diff
@@ -0,0 +1,134 @@
+      LongShortPairFinancialProductLibrary financialProductLibrary; // Contract providing settlement payout logic.
+      bytes customAncillaryData; // Custom ancillary data to be passed along with the price request to the OO.
+      uint256 prepaidProposerReward; // Preloaded reward to incentivize settlement price proposals.
+      uint256 optimisticOracleLivenessTime; // OO liveness time for price requests.
+      uint256 optimisticOracleProposerBond; // OO proposer bond for price requests.
+      FinderInterface finder; // DVM finder to find other UMA ecosystem contracts.
+      address timerAddress; // Timer used to synchronize contract time in testing. Set to 0x000... in production.
+  }
+*/
+
+const HARDHAT_DEFAULT_PRIVATE_KEY =
+  "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+
+export default async function deployLSPContract() {
+  const provider = new ethers.getDefaultProvider("http://127.0.0.1:8545");
+  const signer = new Wallet(HARDHAT_DEFAULT_PRIVATE_KEY, provider);
+  const factoryAddress = getLongShortPairCreatorAddress(1);
+  const lspFactoryAbi = getLongShortPairCreatorAbi();
+
+  const factoryInstance = new ethers.Contract(
+    factoryAddress,
+    lspFactoryAbi,
+    signer
+  );
+
+  const fplAbi = getLongShortPairFinancialProductLibraryAbi();
+
+  // Pulled from networks in uma/core.
+  // Range Financial lib
+  const fplAddress = "0xc1f4e05738E5a7B7CB1f22bB689359CCb1610DA4";
+
+  const fplInstance = new ethers.Contract(fplAddress, fplAbi, signer);
+
+  // LSP parameters. Pass in arguments to customize these.
+  const lspParams = {
+    pairName: LSP_PAIRNAME,
+    expirationTimestamp: 1735718400, // Timestamp that the contract will expire at. exp time: jan 1st, 2025, midnight.
+
+    collateralPerPair: "250000000000000000", // 0.25
+    priceIdentifier: ethers.utils
+      .hexlify(ethers.utils.toUtf8Bytes("UMAUSD"))
+      .padEnd(66, "0"), // Price identifier to use.
+    longSynthName: "UMA Cypress Test Token",
+    longSynthSymbol: "UMACTT-L",
+    shortSynthName: "UMA Short Cypress Test Token",
+    shortSynthSymbol: "UMACTT-S",
+    collateralToken: "0x04Fa0d235C4abf4BcF4787aF4CF447DE572eF828", // Collateral token address, UMA in this case
+    financialProductLibrary: fplAddress,
+    customAncillaryData: "0x", // Default to empty bytes array if no ancillary data is passed.
+    prepaidProposerReward: 0, // Default to 0 if no prepaid proposer reward is passed.
+    optimisticOracleLivenessTime: 7200,
+    optimisticOracleProposerBond: "40000000000000000000",
+    upperBound: "12000000000000000000",
+    lowerBound: "4000000000000000000",
+  };
+
+  console.log("did I get here?");
+
+  let expectedAddress = "";
+  try {
+    expectedAddress = await factoryInstance.callStatic.createLongShortPair(
+      lspParams
+    );
+  } catch (err) {
+    console.log("err in EA call", err);
+  }
+
+  console.log("EA", expectedAddress);
```

Comment:
nit: this log might be confusing if the expected address is an empty string. It also might be nice to make it clearer what it's printing (if this log is meant to be permanent).

Thoughts on moving it into the `try` so it only prints if the expected address is defined?

Reply:
Seems reasonable to me.